components/automate-cli/cmd/chef-automate/ca.go

```diff
@@ -14,56 +16,126 @@ import (
+}
+
 func init() {
-	RootCmd.AddCommand(caCmd())
+	regen.AddCommand(regenRoot)
+	caCmd.AddCommand(caInfo)
+	caCmd.AddCommand(regen)
+
+	caInfo.PersistentFlags().BoolVarP(&caCmdFlags.automate, "automate", "a", false, "Print information of the root certificate of automate nodes")
+	caInfo.PersistentFlags().SetAnnotation("automate", docs.Compatibility, []string{docs.CompatiblewithHA})
+	caInfo.PersistentFlags().BoolVar(&caCmdFlags.automate, "a2", false, "Print information of the root certificate of automate nodes")
+	caInfo.PersistentFlags().SetAnnotation("a2", docs.Compatibility, []string{docs.CompatiblewithHA})
+	caInfo.PersistentFlags().BoolVarP(&caCmdFlags.chef_server, "chef_server", "c", false, "Print information of the root certificate of chef_server nodes")
+	caInfo.PersistentFlags().SetAnnotation("chef_server", docs.Compatibility, []string{docs.CompatiblewithHA})
+	caInfo.PersistentFlags().BoolVar(&caCmdFlags.chef_server, "cs", false, "Print information of the root certificate of chef_server nodes")
+	caInfo.PersistentFlags().SetAnnotation("cs", docs.Compatibility, []string{docs.CompatiblewithHA})
+	caInfo.PersistentFlags().StringVar(&caCmdFlags.node, "node", "", "Node Ip address")
+	caInfo.PersistentFlags().SetAnnotation("node", docs.Compatibility, []string{docs.CompatiblewithHA})
+
+	regenRoot.PersistentFlags().BoolVarP(&caCmdFlags.automate, "automate", "a", false, "Regenerate the root certificate for automate nodes")
+	regenRoot.PersistentFlags().SetAnnotation("automate", docs.Compatibility, []string{docs.CompatiblewithHA})
+	regenRoot.PersistentFlags().BoolVar(&caCmdFlags.automate, "a2", false, "Regenerate the root certificate for automate nodes")
+	regenRoot.PersistentFlags().SetAnnotation("a2", docs.Compatibility, []string{docs.CompatiblewithHA})
+	regenRoot.PersistentFlags().BoolVarP(&caCmdFlags.chef_server, "chef_server", "c", false, "Regenerate the root certificate for chef_server nodes")
+	regenRoot.PersistentFlags().SetAnnotation("chef_server", docs.Compatibility, []string{docs.CompatiblewithHA})
+	regenRoot.PersistentFlags().BoolVar(&caCmdFlags.chef_server, "cs", false, "Regenerate the root certificate for chef_server nodes")
+	regenRoot.PersistentFlags().SetAnnotation("cs", docs.Compatibility, []string{docs.CompatiblewithHA})
+	regenRoot.PersistentFlags().StringVar(&caCmdFlags.node, "node", "", "Node Ip address")
+	regenRoot.PersistentFlags().SetAnnotation("node", docs.Compatibility, []string{docs.CompatiblewithHA})
+
+	RootCmd.AddCommand(caCmd)
 }
 
-func caCmd() *cobra.Command {
-	cmd := &cobra.Command{
-		Use:   "internal-ca COMMAND",
-		Short: "Manage Chef Automate's internal certificate authority",
-		Long:  "Manage Chef Automate's internal certificate authority. Used for inter-service encryption and authentication.",
-		Annotations: map[string]string{
-			docs.Tag: docs.FrontEnd,
-		},
-	}
+func runCAInfoCmd(cmd *cobra.Command, args []string) error {
+	if isA2HARBFileExist() {
+		infoCmd := "sudo " + cmd.CommandPath()
```

Comment:
runCAInfoCmd and runRegenRootCmd have same lines before calling the runInternalCaHA.
Can we put these lines in runInternalCaHA itself?

Reply:
done, but for unit test cases we have to make call to getAutomateHAInfraDetails() function here only.